drivers/snmp-ups.c

```diff
@@ -878,9 +940,10 @@ void su_setinfo(snmp_info_t *su_info_p, const char *value)
 	if (SU_TYPE(su_info_p) == SU_TYPE_CMD)
 		return;
 
-	/* ups.status and ups.alarm have special handling, not here! */
+	/* ups.status and {ups, Lx, outlet, outlet.group}.alarm have special
+	 * handling, not here! */
 	if ((strcasecmp(su_info_p->info_type, "ups.status"))
-		&& (strcasecmp(su_info_p->info_type, "ups.alarm")))
+		&& (strcasecmp(strrchr(su_info_p->info_type, '.'), ".alarm")))
```

Comment:
not sure to fully get your comment here: we just want to process anything that is not "ups.status" or "*.alarm" (i.e. mapping ending with ".alarm, since it's strRchr()), that is "{ups, Lx, outlet, outlet.group}.alarm".
These are not processed (published) using dstate_setinfo(), but using {status,alarm}_{init,set,commit} 

=> grep "\.alarm\"" drivers/*mib*